homeassistant/components/somfy/__init__.py

```diff
@@ -58,6 +60,10 @@
     if DOMAIN not in config:
         return True
 
+    hass.data[DOMAIN][CONF_SHADOW] = False
+    if CONF_SHADOW in config[DOMAIN].keys():
```

Comment:
`if CONF_SHADOW in config[DOMAIN]` must be enough.
https://stackoverflow.com/a/1602964/3715421

Reply:
OK, I'll make this change at the weekend.

Reply:
I've now updated the pull request to switch to the wording "Optimistic"